### PR TITLE
TextWidgetArea typing effect

### PR DESCRIPTION
**Pull request by mishig25**
This PR:
1. add text typing effect to `WidgetTextArea.svelte` component
2. shift+enter to run label to TextGeneration widget

latest demos: [gpt2](https://62c41478139ebb45e36f86a1--huggingface-widgets.netlify.app/gpt2/), [bloom](https://62c41478139ebb45e36f86a1--huggingface-widgets.netlify.app/bigscience/bloom/)

Desktop:

https://user-images.githubusercontent.com/11827707/177184723-775b7179-ecfb-4a9c-9839-e9a9f3cab33d.mov

Mobile:
Screenshot 2022-06-27 at 13 18 27

### CODE REVIEW

File: js/src/lib/components/InferenceWidget/shared/WidgetTextarea/WidgetTextarea.svelte [hunk 1]
```diff
@@ -1,37 +1,55 @@
...
-				return txt;
-			});
-			spanEl.blur();
+	let containerSpanEl: HTMLSpanElement;
+	const typingEffectSpeedMs = 15;
+
+	export async function renderTypingEffect(outputTxt: string) {
+		const spanEl = document.createElement("span");
+		spanEl.contentEditable = "true";
+		spanEl.className = "font-semibold";
```
Comment (mishig25) [inline js/src/lib/components/InferenceWidget/shared/WidgetTextarea/WidgetTextarea.svelte:16]: @gary149 please feel free to modify this line for the generated text style

**Comment by julien-c** [comment]
will this also support:
- styling it in a different color
- having multiple distinct spans of text in different places of the textarea, i.E. being able to type more (human) text and relaunch a generation from any place?

**Comment by mishig25** [comment]
> * having multiple distinct spans of text in different places of the textarea, i.E. being able to type more (human) text and relaunch a generation from any place?

yes

https://user-images.githubusercontent.com/11827707/175941259-f309a6f7-3b55-444f-9283-3e5c6add41af.mov

> * styling it in a different color

This is the line that controls the style of output txt: https://github.com/huggingface/hub-docs/blob/4a6734463ebb138602971fbf1c1a86b403b8c8bd/js/src/lib/components/InferenceWidget/shared/WidgetTextarea/WidgetTextarea.svelte#L16

do you want it to be something like `classNames prop` that can be modified?

**Comment by mishig25** [comment]
latest demosL [gpt2](https://62c41478139ebb45e36f86a1--huggingface-widgets.netlify.app/gpt2/) [bloom](https://62c41478139ebb45e36f86a1--huggingface-widgets.netlify.app/bigscience/bloom/)

**Review by gary149 — APPROVED** [review]

**Pull request merged**